powerline_shell/__init__.py

```diff
@@ -183,8 +196,19 @@ def main():
     powerline = Powerline(args, config, theme)
     segments = []
     for seg_name in config["segments"]:
-        mod = importlib.import_module("powerline_shell.segments." + seg_name)
-        segment = getattr(mod, "Segment")(powerline)
+        segment = None
+        if seg_name is None or len(seg_name)<1:
+            continue                            #ignore bad configs
+        elif type(seg_name) is dict:
+            ret = parseInlineSegment(seg_name)
+            if ret is None:
+                continue                        #ignore bad configs
+            else:
+                mod = importlib.import_module("powerline_shell.segments.inline")
+                segment = getattr(mod, "Segment")(powerline, ret)
+        elif type(seg_name) is str:
+            mod = importlib.import_module("powerline_shell.segments." + seg_name)
+            segment = getattr(mod, "Segment")(powerline)
```

Comment:
All of changes in this file could be drastically simplified. I am thinking along these lines:

```python
for seg_conf in config["segments"]:
    if not isinstance(seg_conf, dict):
        seg_conf = {"type": seg_conf}
    mod = importlib.import_module("powerline_shell.segments." + seg_conf["type"])
    segment = getattr(mod, "Segment")(powerline, seg_conf.get("options"))
```

This should be all that's needed (at least in this file). This does require changing the segment classes to all accept an "options" argument. But you could also avoid that in this PR by changing my last line to:

```python
    if "options" in seg_conf:
        segment = getattr(mod, "Segment")(powerline, seg_conf["options"])
    else:
        segment = getattr(mod, "Segment")(powerline)
```